### PR TITLE
[Merged by Bors] - feat(Util/Superscript): add a semi-correct formatter

### PR DESCRIPTION
**Pull request by eric-wieser**
This probably isn't correct, but the old version was even less correct!

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Util/Superscript.lean [hunk 1]
```diff
@@ -204,15 +207,18 @@ def scriptParser.mapFormat (m : Mapping) (f : Format) : Except String Format :=
...
 def scriptParser.formatter (m : Mapping) (k : SyntaxNodeKind) (p : Formatter) : Formatter := do
   let stack ← modifyGet fun s => (s.stack, {s with stack := #[]})
   Formatter.node.formatter k p
   let s ← get
   match s.stack.mapM <| scriptParser.mapFormat m with
-  | .error s => throwError "Not a superscript: '{s}'"
-  | .ok newStack => set { s with stack := stack ++ newStack }
+  | .error err =>
+    -- TODO: this doesn't seem to go anywhere
+    Lean.logErrorAt (← get).stxTrav.cur s!"Not a superscript: '{err}'"
```
Comment (digama0) [inline Mathlib/Util/Superscript.lean:225]: maybe this should use a trace message instead?

Comment (eric-wieser) [reply Mathlib/Util/Superscript.lean:217]: Do you have a trace class in mind?

Comment (digama0) [reply Mathlib/Util/Superscript.lean:217]: the usual one I enable when formatting is broken is `trace.PrettyPrinter.format.backtrack`, but you could also just use `trace.PrettyPrinter.format`

Comment (eric-wieser) [reply Mathlib/Util/Superscript.lean]: I worked out how to make the error show up after all.

**Pull request closed without merging**